src/model.rs

```diff
@@ -724,6 +724,156 @@ impl Collections {
+    ///  VJ:1   08:00-09:00      10:00-11:00
+    ///  VJ:2                    10:00-11:00      14:00-15:00
+    ///                         X           ▲    |           X
+    ///                         X           |    ▼   SP3     X
+    ///                         out         in   out         in
+    ///                         |- Stay-In -|
+    ///
+    /// Example 2:
+    /// ##########
+    ///       out          in  out               in
+    ///        X    SP1    |    ▲       SP2      X
+    ///        X           ▼    |                X
+    ///  VJ:1   08:00-09:00      10:00---------12:00
+    ///  VJ:2                           11:00----------13:00      13:00-14:00
+    ///                                   X                 ▲    |           X
+    ///                                   X       SP3       |    ▼    SP4    X
+    ///                                 out                 in  out          in
+    ///                         |--------- Stay In ---------|
+    ///
+    /// Note the overlap between the departure time of the last stop point SP2
+    /// of VJ:1 and the arrival time of the first stop point SP3 of VJ:2. In
+    /// this case, we still apply the default rule.
+    ///
+    ///
+    /// Example 3:
+    /// ##########
+    ///       out          in   out         in   out         in   out         in
+    ///        X    SP1    |    ▲    SP2    |    ▲    SP3    |    ▲   SP4     X
+    ///        X           ▼    |           ▼    |           |    |           X
+    ///  VJ:1   08:00-09:00      10:00-11:00     |           ▼    |           X
+    ///  VJ:2                                     12:00-13:00      14:00-15:00
+    ///                         |---------- Stay In ---------|
+    ///
+    /// Example 3 is the only case were we allow specific pick-up and
+    /// drop-off.
+    ///
+    /// Example 4:
+    /// ##########
+    ///                       SP0               SP1               SP2               SP3    
+    ///                                                                                    
+    ///  VJ:1 (Mon-Sun)   09:00-10:00       10:00-11:00                                    
+    ///  VJ:2 (Mon-Fri)                                       12:00-13:00       14:00-15:00
+    ///  VJ:3 (Sat-Sun)                                       12:30-13:30       14:30-15:30
+    ///
+    /// Example 4 might be a valid use case of stay-in but is not handled in the
+    /// current implementation since the validity patterns are different; this
+    /// is undefined behavior (most likely, some stay-in will be forbidden by
+    /// the default rule).
+    pub fn enhance_pickup_dropoff(&mut self) {
+        let mut vj_idxs: Vec<Idx<VehicleJourney>> = self
+            .vehicle_journeys
+            .iter()
+            // Filtering out vehicle journeys without any stop time allowing us
+            // below to access first and last stop_times safely.
+            .filter(|(_, vj)| !vj.stop_times.is_empty())
```

Comment:
Would it be better to filter on length <= 1 ?

Reply:
Technically, with a vehicle journey containing only one `stop_time`, I believe we should forbid pick-up and drop-off on this only stop and pretty much make the vehicle journey unusable.

Now, if we filter with `.len() <= 1` as you propose, then we won't forbid pick-up and drop-off on this weird vehicle journey with the more risk it end up being showed up to a traveler? In terms of traveler information, I'd think we would prefer to forbid at maximum this kind of weird vehicle journey?